include/xhyve/support/misc.h

```diff
@@ -65,3 +68,25 @@ static inline void do_cpuid(unsigned ax, unsigned *p) {
 
 /* Used to trigger a self-shutdown */
 extern void push_power_button(void);
+
+/* Error checking pthread mutex operations */
+static inline void xpthread_mutex_init(pthread_mutex_t *mutex)
+{
+	int rc = pthread_mutex_init(mutex, NULL);
+	assert(rc == 0);
```

Comment:
Shouldn't this be a dynamic check that called `xhyve_abort`, since there are valid non-zero return codes from `pthread_mutex_init`?  E.g. `ENOMEM`.  Asserts won't fire if disabled.